src/components/dashboard/Mentorship.js

```diff
@@ -48,27 +49,34 @@ class Mentorship extends React.Component {
       all: true,
     }
   }
-  
+
+  componentWillUnmount() {
+    /*** We would like to store filter state in Redux on component unmounting
+     * (when user navigates away), and restore it from state when the component
+     * is mounted again (user navigates back)— thoughts?
+    */
+  }
```

Comment:
@no-stack-dub-sack after adding the links to create private chat I found myself returning to the mentorship page a lot and feeling like it would be a nice experience if the filters and search were restored. Mainly because users may easily navigate away (to contact a mentor for instance) and there are a lot of search options so it could be harder to recreate a complicated search state. Besides, since we have single-page-application awesomeness I think we should maintain the state and instead just provide a button to clear all the search fields, additionally, this is currently the behavior on all the other routes. But we can discuss further.